src/pymodaq/pid/pid_controller.py

```diff
@@ -301,17 +299,19 @@ def stop_moves(self, overshoot):
         for mod in self.modules_manager.actuators:
             mod.stop_Motion()
 
-    def set_model(self):
-        model_name = self.settings.child('models', 'model_class').value()
-        self.model_class = find_dict_in_list_from_key_val(self.models, 'name', model_name)['class'](self)
-        self.set_setpoints_buttons()
-        self.model_class.ini_model()
-        self.settings.child('main_settings', 'epsilon').setValue(self.model_class.epsilon)
-
+    @Slot()
     def ini_model(self):
+        """Initialize the model.
+
+        Called by the UI "Ini model" button.
+        Construct the UI depending on the number of setpoints defined in the model.
+        Call the ini_model function of the model.
+        """
         try:
-            if self.model_class is None:
-                self.set_model()
+            model_name = self.settings.child('models', 'model_class').value()
+            self.model_class = find_dict_in_list_from_key_val(self.models, 'name', model_name)['class'](self)
+            self.set_setpoints_buttons()
+            self.model_class.ini_model()
```

Comment:
if I removed it from there there should be a reason...well maybe I did crap, but that could cause some hidden bug...

Reply:
Yes actually I do not know so i put it back as it was.

Reply:
I put it back as it was.